pkg/operator/rgw/rgw.go

```diff
@@ -49,236 +46,177 @@ const (
 	certFilename   = "rgw-cert.pem"
 )
 
-// Cluster for rgw management
-type Cluster struct {
-	context   *clusterd.Context
-	Namespace string
-	placement k8sutil.Placement
-	Version   string
-	Config    model.ObjectStore
-}
-
-type idType struct {
-	ID string `json:"id"`
-}
-
-type realmType struct {
-	Realms []string `json:"realms"`
-}
-
-// New creates an instance of an rgw manager
-func New(context *clusterd.Context, config model.ObjectStore, namespace, version string, placement k8sutil.Placement) *Cluster {
+// Start the rgw manager
+func (s *ObjectStore) Create(context *clusterd.Context, version string) error {
 
-	return &Cluster{
-		context:   context,
-		Namespace: namespace,
-		placement: placement,
-		Version:   version,
-		Config:    config,
+	// validate the object store settings
+	if err := s.validate(); err != nil {
+		return fmt.Errorf("invalid object store %s arguments. %+v", s.Name, err)
 	}
-}
 
-// Start the rgw manager
-func (c *Cluster) Start() error {
-	logger.Infof("start running rgw")
-
-	err := c.createPools()
-	if err != nil {
-		return fmt.Errorf("failed to create pools. %+v", err)
+	// check if the object store already exists
+	exists, err := s.exists(context)
+	if err == nil && exists {
+		logger.Infof("object store %s already exists in namespace %s ", s.Name, s.Namespace)
+		return nil
 	}
 
-	err = c.createKeyring()
+	logger.Infof("creating object store %s in namespace %s", s.Name, s.Namespace)
+	err = s.createKeyring(context)
 	if err != nil {
 		return fmt.Errorf("failed to create rgw keyring. %+v", err)
 	}
 
 	// start the service
-	serviceIP, err := c.startService()
+	serviceIP, err := s.startService(context)
 	if err != nil {
 		return fmt.Errorf("failed to start rgw service. %+v", err)
 	}
 
-	err = c.createRealm(serviceIP)
+	// create the ceph artifacts for the object store
+	objContext := cephrgw.NewContext(context, s.Name, s.Namespace)
+	err = s.createObjectStore(objContext, serviceIP)
 	if err != nil {
-		return fmt.Errorf("failed to create realm. %+v", err)
+		return fmt.Errorf("failed to create ceph object store. %+v", err)
+	}
+
+	// start the deployment or daemonset
+	var rgwType string
+	if s.Spec.Gateway.AllNodes {
```

Comment:
is there an use case not to use allNodes for RGW?

Reply:
the real use case will be to auto-scale rgw depending on load, but we'll save that for another PR